2022/incoming/create-your-excel-using-perl.pod

```diff
@@ -0,0 +1,170 @@
+Let's add some more format objects for later.
+
+=begin code
+
+	my $format2 = $merry->add_format();
+
+	#Setting format using formatting methods, red and bold text, green background, center alignment, green background and dotted borders (for the snowflakes)
+
+	$format2->set_color('red');
+	$format2->set_bold();
+	$format2->set_size( 12 );
+	$format2->set_bg_color( 'green' );
+	$format2->set_align('center');
+	$format2->set_border( 4 );
+
+	#Just some snowflakes as border
+	my $format3 = $merry->add_format( border => 4,);
+
+=end code
+
+=head2 Writing data into cells
+
+There are two ways to do that: You can use either the row-column notation or the A1 notation as described in L<Cell notation|https://metacpan.org/pod/Excel::Writer::XLSX#Cell-notation>.
+First of all we need some data, let's compose our cookies list.
+We will use a hash of array refs to do this:
+
+=begin code
+
+	my $cookies = { 
+		      Hungary => [ 'bejgli', 'hólabda' ],
+		      Scandinavia => [ 'Pepparkakor', 'Ruiskakut', 'Lussekatter', 'Krumkake' ],
+		      Netherlands => [ 'Spekulaas', 'Kerstkranjes'],
+		      Germany => [ 'Lebkuchen', 'Stollen' ],
+		      Switzerland => [ 'Basler Läckerli', 'Brunsli' ],
+		      Italy => [ 'Panettone', 'Pignoli', 'Cannoli' ],
+		      Spain => [ 'Polvorones', 'Mantecados', 'Turrón' ],
+		      Greece => [ 'Melomakarona' ],
+		      };
+		      
+	#Sort the countries alphabetically
+	my @countries = sort keys %$cookies;
+
+	#Add the countries to the baking sheet, one country per column.
+	$baking_sheet->set_column( 0, scalar keys %$cookies, 15);
+	$baking_sheet->write( 'A1', \@countries, $format2 );
+
+	my $i = 0;
+	for ( sort keys %$cookies ) {
+	  #Place Your cookies to the baking sheet
+	  $baking_sheet->write( 1, $i, [ $cookies->{$_} ], $format3 );
+	  $i++;
+	}
+
+	#write some statistics about how many types of cookies we have per country
+	$stats->write( 'A1', [\@countries], $format2 );
+
+	#Reset the counter
+	$i = 0;
+	#count how many cookies we country wise.
+	for ( sort keys %$cookies ) {
+	  #Place Your cookies to the baking sheet
+	  $stats->write( $i, 1, scalar @{$cookies->{$_}} , $format3 );
+	  $i++;
+	}
+        
+=end code
+
+Many things are happening here.
+
+First we get a list of countries sorted alphabetically.
+Second we are writing those countries into the first row, alongside some formats, defined above.
+Then comes the first usage of C<write()>, where we are using the A1 notation to start from the top left corner, then comes the reference to the C<@countries> array, followed by the format2.
+When we pass an array reference to C<write()>, it will write the elements of the array in the same row. It is equivalent to write_row().
+If You pass and array ref of array refs to C<write()>, it is the same if You would call C<write_col()>, it will write to the columns, as demonstrated in the lines 22 and 27.
```

Comment:
s/pass and array/pass an array/
s/same if You would call/same as calling/

Reply:
done